pathod/language/base.py

```diff
@@ -376,6 +376,30 @@ def freeze(self, settings):
         return self.__class__(self.value.freeze(settings))
 
 
+class BytesWrapper():
```

Comment:
I agree that the byte handling is a bit fiddly, but I'm not sure if introducing a second kind of `bytes` with improved/different semantics is really worth it. For example, you cannot call `.decode()` on this anymore, etc. etc..

I'd rather have us do the `[i:i+1]` dance where required - I believe/hope we have tackled most problem instances of this anway.